angular-app/src/app/app-routing.module.ts

```diff
@@ -1,55 +1,38 @@
 import { NgModule } from '@angular/core';
-import { RouterModule, Routes } from '@angular/router';
+import { Routes, RouterModule, UrlSegment, UrlMatchResult, RouteReuseStrategy } from '@angular/router';
 import { MainContentComponent } from "./components/main-content/main-content.component";
+import { AemPageDataResolver, AemPageRouteReuseStrategy } from "@adobe/cq-angular-editable-components";
 
-import { environment } from '../../src/environments/environment';
-
-const CONTEXT_PATH = environment.CONTEXT_PATH;
+export function AemPageMatcher(url: UrlSegment[]): UrlMatchResult {
+  if (url.length) {
+    return (
+      {
+        consumed: url,
+        posParams: {
+          path: url[url.length - 1]
+        }
+      }
+    );
+  }
+}
 
 const routes: Routes = [
-    {
-      path: CONTEXT_PATH + 'content/we-retail-journal/angular/blog.html',
-      component: MainContentComponent,
-      data: {
-        path: '/' + CONTEXT_PATH + 'content/we-retail-journal/angular/blog'
-      }
-    },
-    {
-      path: CONTEXT_PATH +'content/we-retail-journal/angular/blog/aboutus.html',
-      component: MainContentComponent,
-      data: {
-        path: '/' + CONTEXT_PATH + 'content/we-retail-journal/angular/blog/aboutus'
-      }
-    },
-    {
-      path: CONTEXT_PATH +'content/we-retail-journal/angular/blog/weather.html',
-      component: MainContentComponent,
-      data: {
-        path: '/' + CONTEXT_PATH + 'content/we-retail-journal/angular/blog/weather'
-      }
-    },
-    {
-      path: CONTEXT_PATH +'content/we-retail-journal/angular/home.html',
-      component: MainContentComponent,
-      data: {
-        path: '/' + CONTEXT_PATH + 'content/we-retail-journal/angular/home'
-      }
-    },
-    {
-      path: CONTEXT_PATH +'content/we-retail-journal/angular.html',
-      redirectTo: '/' + CONTEXT_PATH + 'content/we-retail-journal/angular/home.html'
-    },
-    {
-      path: '',
-      redirectTo: '/' + CONTEXT_PATH + 'content/we-retail-journal/angular.html',
-      pathMatch: 'full'
+  {
```

Comment:
Does this handle the initial "blank" path? Ie. http://localhost:4502 ? I'd expect that an empty path would automatically redirect to the root page of the app (else its not obvious why nothing is being show)

WDYT?

Reply:
Catches for blank page or catch-all are missing in routing definition, as at the time of setting up routing we don't know yet the page model, thus we don't know what would be the redirection target.

However, in the root content of angular and react we have added `cq:redirectTarget` that would redirect to desired target that would handle requests like http://localhost:4502/editor.html/content/we-retail-journal/angular.html for example. Later we might add root redirection that would also handle blank url case.

Reply:
It looks like developers could run `ng serve` with in tandem with the `open` command:

`ng serve && open http://localhost:4200/content/we-retail-journal/angular/home.html`  

to avoid an initial blank load (or whatever the "home page" is). 

Of course they could optionally create an` package.json` script entry for this as well (`start:localdev`) or something, and sync it up w/ whatever is in their index.html as the `cq:root_model_path`